camel/file_io/unstructured_io.py

```diff
@@ -0,0 +1,411 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+
+
+class UnstructuredModules:
+    UNSTRUCTURED_MIN_VERSION = "0.10.30"  # Define the minimum version
+
+    def ensure_unstructured_version(self, min_version: str) -> None:
+        r"""Validates that the installed 'Unstructured' library version
+        satisfies the specified minimum version requirement. This function is
+        essential for ensuring compatibility with features that depend on a
+        certain version of the 'Unstructured' package.
+
+        Args:
+            min_version (str): The minimum version required,
+            specified in 'major.minor.patch' format.
+
+        Raises:
+            ImportError: If the 'Unstructured' package
+            is not available in the environment.
+            ValueError: If the current 'Unstructured'
+            version is older than the required minimum version.
+
+        Notes:
+            Uses the 'packaging.version' module to parse
+            and compare version strings.
+        """
+        from packaging import version
+
+        try:
+            from unstructured.__version__ import __version__
+
+        except ImportError as e:
+            raise ImportError("Package `unstructured` not installed.") from e
+
+        # Use packaging.version to compare versions
+        min_ver = version.parse(min_version)
+        installed_ver = version.parse(__version__)
+
+        if installed_ver < min_ver:
+            raise ValueError(f"Require `unstructured>={min_version}`, "
+                             f"you have {__version__}.")
+
+    def parse_file_or_url(self, input_path: str) -> Any:
```

Comment:
Should me have an argument `return_str` and return `str` by default? That may be easy for the agent to use it a tool. Also it would be easier for the users who are not familiar with `unstructured`.

Reply:
great point, have added the argument